packages/babel-parser/src/parser/expression.js

```diff
@@ -532,10 +532,6 @@ export default class ExpressionParser extends LValParser {
       return expr;
     }
 
-    if (this.checkExpressionErrors(refExpressionErrors, false)) {
```

Comment:
We should investigate if the other `this.checkExpressionErrors(refExpressionErrors, false)` checks is necessary. They are not caught by any test cases in babel-parser now.

(They are not even caught by acorn testcases)

Reply:
I was thinking of removing those checks, but I wasn't sure if I should do that in this PR, since removing those checks wouldn't be necessary to fix this specific bug.

I will look into removing these checks.

Reply:
I will look into removing the other instances of `this.checkExpressionErrors(refExpressionErrors, false)`. I didn't remove them in this PR, because it wasn't necessary to fix the bug, but I can always add some more changes to this PR/open a new one.